### PR TITLE
small_upgrade_fixes

### PR DESCRIPTION
**Pull request by jyejare**
This contains fix for:
1. Port 22 low level socket error on openstack instance: - Polling with timeout added 
2. I dont know how but the Timeout improvements from PR #232 has been disappeared from code so re-added.
3. Changed the way of importing Client from novaclient module as per jenkins warning - Removed version 'v2' while importing Client. It works :)

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1922,20 +1923,39 @@ def host_pings(host, attempts=200):
...
     """Retrives the hostname by logging into remote machine by IP.
     Specially for the systems who doesnt support reverse DNS.
     e.g usersys machines.
 
     :param ip: A string. The IP address of the remote host.
+    :param timeout: An int. The polling timeout to get the hostname using IP.
 
     """
-    return execute(lambda: run('hostname'), host=ip)[ip]
+    timeup = time.time() + int(timeout)*60
```
Comment (elyezer) [inline automation_tools/__init__.py]: Please add a space between the * operator.

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -1922,20 +1923,39 @@ def host_pings(host, attempts=200):
...
 
 
-def get_hostname_from_ip(ip):
+def get_hostname_from_ip(ip, timeout=3):
     """Retrives the hostname by logging into remote machine by IP.
     Specially for the systems who doesnt support reverse DNS.
     e.g usersys machines.
 
     :param ip: A string. The IP address of the remote host.
+    :param timeout: An int. The polling timeout to get the hostname using IP.
```
Comment (elyezer) [inline automation_tools/__init__.py]: `:param int timeout: The polling timeout in seconds`

**Comment by elyezer** [comment]
ACK, pending comments.

**Comment by sghai** [comment]
ACK, pending on @elyezer's comments..

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -1922,20 +1922,39 @@ def host_pings(host, attempts=200):
...
 
 
-def get_hostname_from_ip(ip):
+def get_hostname_from_ip(ip, timeout=3):
     """Retrives the hostname by logging into remote machine by IP.
     Specially for the systems who doesnt support reverse DNS.
     e.g usersys machines.
 
     :param ip: A string. The IP address of the remote host.
+    :param timeout: An integer. The polling timeout in minutes.
```
Comment (sghai) [inline automation_tools/__init__.py]: I think elyezer's comments still needs to be addressed.
`:param int timeout: The polling timeout in minutes`

Please check other instances too.

Comment (jyejare) [inline automation_tools/__init__.py]: All over the upgrade automation, I have followed current style and need to change it all over.
So I will be raising new PR for that. As of now you may merge this PR.

Comment (elyezer) [inline automation_tools/__init__.py]: Please update to `:param int timeout: ...` like previous comment.

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -1989,6 +2009,7 @@ def create_openstack_instance(
         e.g m1.small.
     :param ssh_key: A string. ssh_key 'name' that required to add
         into this instance.
+    :param timeout: An integer.The polling timeout in minutes to assign IP.
```
Comment (elyezer) [inline automation_tools/__init__.py]: Please update to `:param int timeout: ...` like previous comment.

File: automation_tools/__init__.py [hunk 5]
```diff
@@ -1904,15 +1904,15 @@ def sync_capsule_tools_repos_to_upgrade(admin_password=None):
...
 
 
-def host_pings(host, attempts=200):
+def host_pings(host, timeout=15):
     """This ensures the given IP/hostname pings succesfully.
 
     :param host: A string. The IP or hostname of host.
-    :param attempts: An integer.
-        The maximum number of attempts to ping the host.
+    :param timeout: An integer. The polling timeout in minutes.
```
Comment (elyezer) [inline automation_tools/__init__.py]: Missing here.

Comment (jyejare) [inline automation_tools/__init__.py]: OMG !!

**Comment by elyezer** [comment]
APT

**Pull request merged**